pkgs/by-name/sp/spacetimedb/package.nix

```
@@ -0,0 +1,49 @@
+{
+  lib,
+  fetchFromGitHub,
+  rustPlatform,
+  pkg-config,
+  perl,
+  git,
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "spacetimedb";
+  version = "1.0.0";
+
+  src = fetchFromGitHub {
+    owner = "clockworklabs";
+    repo = "spacetimedb";
+    tag = "v${version}";
+    hash = "sha256-L3D7DfMQNuoZ/twAsrK20royIGp6PXCAFZKnb0PgSu0=";
+  };
+
+  useFetchCargoVendor = true;
+  cargoHash = "sha256-eOZRp3LRbQzHfT+evKY55ifevX+ki9oT5B7vZs3ym+c=";
+
+  nativeBuildInputs = [
+    pkg-config
+    perl
+  ];
+
+  # Replace hardcoded git binary
+  postPatch = ''
+    substituteInPlace crates/cli/build.rs --replace-fail 'Command::new("git")' 'Command::new("${lib.getExe git}")'
+  '';
+
+  checkFlags = [
+    # requires wasm32-unknown-unknown target
+    "--skip=codegen"
+  ];
+
+  postInstall = ''
```

You can also add a version check after installation:

```suggestion
  ];
  
  doInstallCheck = true;
  nativeInstallCheckInputs = [ versionCheckHook ];
  versionCheckProgramArg = "--version";

  postInstall = ''
```